enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/replication/RaftReplicator.java

```diff
@@ -80,13 +74,13 @@ public RaftReplicator( LeaderLocator leaderLocator, MemberId me,
         do
         {
             assertDatabaseNotShutdown();
-            outbound.send( leader, new RaftMessages.NewEntry.Request( me, operation ) );
             try
             {
+                outbound.send( leaderLocator.getLeader(), new RaftMessages.NewEntry.Request( me, operation ) );
                 progress.awaitReplication( timeout.getMillis() );
                 timeout.increment();
             }
-            catch ( InterruptedException e )
+            catch ( InterruptedException | NoLeaderFoundException e )
```

Comment:
I think it would be good if we kept retrying here, to keep up exactly-once behaviour as far as possible. So the NoLeader case can be handled by just ignoring or logging it perhaps. You could do this by putting the send() in it's own try-catch. We clearly have no test catching this, so at the very least we should add a comment about the reasoning here.